### DISCUSSION TITLE
Not able to preview data on CKAN GUI when uploaded using link

### DISCUSSION THREAD

**Discussion by ArqamFarooqui110719** [ckan/ckan#6947] [Q&A]
**CKAN version** 
CKAN-2.7 (Docker) 

**Describe the bug**
I am not able to preview data on CKAN GUI when uploaded using link.

**Steps to reproduce**

1. Uploaded the new resource using link (https://jsonplaceholder.typicode.com/todos) as below:

![resource_upload_using_link](https://user-images.githubusercontent.com/55977298/175947726-07ef8f9f-6f88-4685-954c-7113dc59d228.png)

2. After successful upload, when I'm previewing the uploaded data, it is showing as below:

![data_preview_uploaded_by_link](https://user-images.githubusercontent.com/55977298/175947731-fcd860ac-ab78-4c48-acaf-b761145f85b7.png)

**Expected behavior**
The uploaded data should be visible on CKAN preview Page. 

Is the above behavior is a bug on CKAN?
Do I need to enable any specific plugin to preview the uploaded data?
Could you please confirm the above queries. 

Thanks you:)

**Comment by amercader** [comment]
To be able to visualize resource you need to install and enable the relevant view plugins. For json resources you might want to use the included `text_view` plugin. Here is more info about resource views: http://docs.ckan.org/en/latest/maintaining/data-viewer.html

**Reply by ArqamFarooqui110719** [reply]
Hi @amercader , thanks for your response.
I had already enable the `text_view` plugin in the ckan configuration file i.e. ckan.ini. The screenshot is attached as below:

![ckan_ini_config](https://user-images.githubusercontent.com/55977298/176385702-960b40c8-19bc-4aa6-bb1d-f2dce69d4933.png)

We have also enable the `text_view` plugin in `ckan_plugins` section i.e.
```
ckan.plugins = stats text_view image_view recline_view datastore oauth2 ngsiview resource_proxy ...
```
and the above text_view plugin is working when we are uploading `.json` file, please find attached screenshot for the same:
![json_file_preview_using_text_view](https://user-images.githubusercontent.com/55977298/176388399-65bb663b-fa89-41ec-ba86-991d225bd7bf.png)

**Reply by amercader** [reply]
In order for the text_view plugin to pick up your resource it needs to have a `JSON` format defined. When you upload a file that is automatically picked up, but when you are linking to an external URL you need to enter it manually when creating or editing the resource
